static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js

```diff
@@ -0,0 +1,137 @@
+                        };
+                    })(f);
+                }
+            }
+        })(
+            document,
+            window.permutive,
+            'd6691a17-6fdb-4d26-85d6-b3dd27f55f08',
+            '359ba275-5edd-4756-84f8-21a24369ce0b',
+            {}
+        );
+        (window.googletag = window.googletag || {}),
+            (window.googletag.cmd = window.googletag.cmd || []),
+            window.googletag.cmd.push(() => {
+                if (
+                    window.googletag.pubads().getTargeting('permutive')
+                        .length === 0
+                ) {
+                    const g = window.localStorage.getItem('_pdfps');
+                    window.googletag
+                        .pubads()
+                        .setTargeting('permutive', g ? JSON.parse(g) : []);
+                }
+            });
+        /* eslint-enable */
+        try {
+            const {
+                isPaidContent,
+                pageId,
+                headline,
+                contentType,
+                section,
+                author,
+                keywords,
+                webPublicationDate,
+                series,
+            } = config.get('page');
+
+            const safeAuthors =
+                author && typeof author === 'string' ? author.split(',') : null;
+            const safeKeywords =
+                keywords && typeof keywords === 'string'
+                    ? keywords.split(',')
+                    : null;
+            const safePublishedAt =
+                webPublicationDate && typeof webPublicationDate === 'number'
+                    ? new Date(webPublicationDate).toISOString()
+                    : null;
+            const rawPayload = {
+                premium: isPaidContent,
+                id: pageId,
+                title: headline,
+                type: contentType,
+                section,
+                authors: safeAuthors,
+                keywords: safeKeywords,
+                publishedAt: safePublishedAt,
+                series,
+            };
+
+            const isEmpty = value =>
+                value === '' || typeof value === 'undefined' || value === null;
+
+            const removeEmpty = payload => {
+                Object.keys(payload).forEach(
+                    key => isEmpty(payload[key]) && delete payload[key]
+                );
+                return payload;
+            };
+
+            const payload = removeEmpty(rawPayload);
+            permutive.addon('web', {
+                page: {
+                    content: payload,
+                },
+            });
+        } catch (err) {
+            reportError(err, { feature: 'comercial' }, false);
+        }
+
+        return resolve();
+    });
+};
+
+export const initPermutiveLib = (): Promise<void> => {
+    const setupPermutive = (): Promise<void> =>
+        loadScript(
+            '//cdn.permutive.com/d6691a17-6fdb-4d26-85d6-b3dd27f55f08-web.js',
```

Comment:
We can maybe extract permutive url  into an aws parameter, as we similarly do for `googletagJsUrl`

Reply:
Funnily enough initially I did extract it out as you say, but I changed to the way the other third party tags are referenced from the code itself (facebook, IMR, Outbrain etc..)
In a way, this isn't something that we'd expect to change often and it might make it easier for the next developer to change it (if it happens at all) from within the code.

Reply:
sounds good to me 👍

Reply:
I agree with Ioanna's initial point. This URL includes an API key and we should avoid having those in the code. Parameter store would be better for this.

Reply:
External resource on this: https://www.freecodecamp.org/news/how-to-securely-store-api-keys-4ff3ea19ebda/

Reply:
@ripecosta all the other third party tags are loaded this way within the codebase (facebook, IMR, twitter, simplereach, plista, outbrain) in the `static/src/javascripts/projects/commercial/modules/third-party-tags` folder they're all there alongside permutive. the API keys are public and they end up on the client anyway!

Reply:
Nice link though! didn't know about `git-secret` and `git-crypt` 🙌

Reply:
Keep in mind this API key is (by design) publicly available, since it is for use in our clientside integration. It's a different class of problem to secret API keys where the mere possession of that secret grants special powers or access.